### PR TITLE
Add Python API usage example

### PR DESCRIPTION
**Pull request by ssbarnea**
Fixes: #297

### CODE REVIEW

File: docs/development.rst [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+
+Basic example of running the linter from python:
+
+.. code-block:: python
+
+   import yamllint
+
+   yaml_config = yamllint.config.YamlLintConfig("extends: default")
+   for p in yamllint.linter.run(".travis.yml", yaml_config):
+         print(p.desc, p.line, p.rule)
```
Comment (adrienverge) [inline docs/development.rst:15]: 6-space indentation is quite uncommon, can you please use 4 (the default for most Python projects)?

File: docs/development.rst [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+to write your own linting tools:
+
+Basic example of running the linter from python:
+
+.. code-block:: python
+
+   import yamllint
+
+   yaml_config = yamllint.config.YamlLintConfig("extends: default")
+   for p in yamllint.linter.run(".travis.yml", yaml_config):
```
Comment (adrienverge) [inline docs/development.rst:14]: Maybe `example.yaml` would be easier to understand for new users?
And still for consistency, can you please use the `.yaml` extension?

File: docs/development.rst (hunk 1 above)
Comment (ssbarnea) [reply docs/development.rst:15]: LOL, vscode got confused by the rst 3 dots prefix. Fix.

File: docs/development.rst [hunk 3]
```diff
@@ -1,8 +1,18 @@
 Development
 ===========
 
-yamllint provides both a script and a Python module. The latter can be used to
-write your own linting tools:
+``yamllint`` provides both a script and a Python module. The latter can be used
+to write your own linting tools:
+
+Basic example of running the linter from python:
```
[the remark this answers is not in the document]
Comment (ssbarnea) [reply docs/development.rst:7]: done

File: docs/development.rst [hunk 4]
```diff
@@ -1,8 +1,18 @@
 Development
 ===========
 
-yamllint provides both a script and a Python module. The latter can be used to
-write your own linting tools:
+``yamllint`` provides both a script and a Python module. The latter can be used
+to write your own linting tools:
```
[the remark this answers is not in the document]
Comment (ssbarnea) [reply docs/development.rst:5]: done

File: docs/development.rst [hunk 5]
```diff
@@ -1,8 +1,18 @@
 Development
 ===========
 
-yamllint provides both a script and a Python module. The latter can be used to
-write your own linting tools:
+yamllint provides both a script and a Python module. The latter can be used
+to write your own linting tools:
```
Comment (adrienverge) [inline docs/development.rst:5]: Could you just restore this text that you don't change?

**Review by adrienverge — COMMENTED** [review]
Looks good, just one thing.

**Review by adrienverge — APPROVED** [review]
Thanks for your contribution @ssbarnea.

**Pull request merged**